### PR TITLE
Feat front end auth

### PR DESCRIPTION
**Pull request by ewansheldon**

### CODE REVIEW

File: src/api/cardsApi.js [hunk 1]
```diff
@@ -1,24 +1,16 @@
...
-function handleResponse(response) {
-  if (response.ok) return response.json();
-  throw new Error(response.text());
-}
+import fetchWrapper from './fetchApi';
 
 function invalid(card) {
   return (!card.text || !card.text.trim().length
-      || card.columnId === null || card.columnId === undefined);
+        || card.columnId === null || card.columnId === undefined);
```
Comment (TomSpencerLondon) [inline src/api/cardsApi.js:5]: Should we extract these boolean conditions into separate methods?

File: src/api/fetchApi.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  headers = { 'Content-Type': 'application/json', Authorization: token },
+  body
+} = {}) {
+  const payload = { method, headers };
+  if (body) {
+    payload.body = JSON.stringify(body);
+  }
+  const response = await fetch(process.env.API_URL + endpoint, payload);
+  return handleResponse(response);
+}
```
Comment (TomSpencerLondon) [inline src/api/fetchApi.js:24]: Great! I like the use of this fetchWrapper method.

File: webpack.config.dev.js [hunk 3]
```diff
@@ -9,7 +9,8 @@ module.exports = {
   entry: './src/index.js',
   plugins: [
     new webpack.DefinePlugin({
-      'process.env.API_URL': JSON.stringify('http://localhost:5000')
+      'process.env.API_URL': JSON.stringify('http://localhost:5000'),
+      'process.env.GOOGLE_AUTH_CLIENT_ID': JSON.stringify(process.env.GOOGLE_AUTH_CLIENT_ID)
```
Comment (TomSpencerLondon) [inline webpack.config.dev.js:13]: These keys for don't need to have process.env I believe. It is correct usage in the values. As discussed.

File: webpack.config.prod.js [hunk 4]
```diff
@@ -18,7 +18,8 @@ module.exports = {
   plugins: [
     new webpack.DefinePlugin({
       'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV),
-      'process.env.API_URL': JSON.stringify('http://retropolisbe-env.eba-xifiu248.eu-west-2.elasticbeanstalk.com')
+      'process.env.API_URL': JSON.stringify('http://retropolisbe-env.eba-xifiu248.eu-west-2.elasticbeanstalk.com'),
+      'process.env.GOOGLE_AUTH_CLIENT_ID': JSON.stringify(process.env.GOOGLE_AUTH_CLIENT_ID)
```
Comment (TomSpencerLondon) [inline webpack.config.prod.js:22]: And again here.

**Review by TomSpencerLondon — APPROVED** [review]

**Review by apavlidi — APPROVED** [review]
Awesome job guys!

**Pull request merged**